R/requirements_helpers.R

```diff
@@ -178,17 +236,17 @@ install_reqs = function(reqs, dryrun, verbose = dryrun,
         v = vmess(sprintf(NOT_INSTALLED,i),verbose)
         install_needed = TRUE
       } else {
-        install_needed = !check_version(version,installed[i],comp)
+        install_needed = !compare_version(installed[i], version, comp)
       }
       if(install_needed){
         if(!is.na(installed[i])) v = vmess(sprintf(BAD_VERSION,package,installed[i]),verbose)
 
         available_versions = get_available_versions(package)
-        available_compatibility = vapply(available_versions,
-                                         check_version,
-                                         TRUE,
-                                         target = version,
-                                         comp = comp)
+
+        available_compatibility = vapply(available_versions, function(available_version) {
+          compare_version(available_version, version, comp)
+        }, logical(1))
+
```

Comment:
Is there a reason why you're using an anonymous function instead of just passing in coompare_version with explicitly named arguments?

Reply:
I personally find it more readable but I’m open to either way. Probably a hint slower to use the anonymous function.